api/sys_init.go

```diff
@@ -19,14 +27,19 @@ func (c *Sys) InitStatus() (bool, error) {
 }
 
 func (c *Sys) Init(opts *InitRequest) (*InitResponse, error) {
+	return c.InitWithContext(context.Background(), opts)
+}
+
+func (c *Sys) InitWithContext(ctx context.Context, opts *InitRequest) (*InitResponse, error) {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
 	r := c.c.NewRequest("PUT", "/v1/sys/init")
```

Comment:
I notice in several places you're replacing "GET" with http.MethodGet, but I don't see it for the PUTs here

Reply:
I wanted to do another PR to go over all of the vault/api methods and replace the strings with constants. I merged the ones from Vinny's suggestions only for the already modified lines.

Reply:
On a second thought, I decided to revert the http.MethodGet/Put changes and will do them in a separate PR.